src/common/tusb_compiler.h

```diff
@@ -75,7 +75,11 @@
  * Nth position is the same as the number of arguments
  * - ##__VA_ARGS__ is used to deal with 0 paramerter (swallows comma)
  *------------------------------------------------------------------*/
-#define TU_ARGS_NUM(...) 	 _TU_NARG(_0, ##__VA_ARGS__,_RSEQ_N())
+#if !defined(__CCRX__)
+#define TU_ARGS_NUM(...)   _TU_NARG(_0, ##__VA_ARGS__,_RSEQ_N())
+#else
+#define TU_ARGS_NUM(...)   _TU_NARG(_0, __VA_ARGS__,_RSEQ_N())
```

Comment:
just for clarification, could ccrx automatically swallow the extra comma when `__VA_ARGS__` expanded as no arguments ?

Reply:
No, it generates a compiler error if no argument is passed. Is it intended to call this macro without an argument ?

Reply:
Thanks for clarification, it is not quite intended to use without arguments, but the macro is for counting number of arguments. So it is more like it will work from 0 to n. Luckily, currently there is no usage for counting zero arguments, but it may (or not) be needed in the future. 

If the CCRX does not support this extension, we could just merge this as it is, no problem at all.

Reply:
@Wini-Buh would you confirm that `##__VA_ARGS__` as `swallowing semicolon with zero-argument` is not supported on CCRX

Reply:
Calling this macro with an argument `TU_CHECK_MCU(OPT_MCU_RX72N)` :

with `_TU_NARG(_0, ##__VA_ARGS__,_RSEQ_N())` sub-macro
 -> compiles with an error with the CCRX toolchain

with `_TU_NARG(_0, __VA_ARGS__,_RSEQ_N())` sub-macro
 -> compiles **successful** with the CCRX toolchain
 
 
Calling this macro without an argument TU_CHECK_MCU() 

with `_TU_NARG(_0, ##__VA_ARGS__,_RSEQ_N())` sub-macro
 -> compiles with an error with the CCRX toolchain
 
with `_TU_NARG(_0, __VA_ARGS__,_RSEQ_N())` sub-macro
 -> compiles with an error with the CCRX toolchain

@hathach  I hope this helps you further.

Reply:
I see, apparently `##__VA_ARGS__` is not supported by ccrx